src/ReinforcementLearningZoo/src/algorithms/policy_gradient/A2C.jl

```diff
@@ -60,15 +60,15 @@ function _update!(learner::A2CLearner, t::CircularArraySARTTrajectory)
     w₃ = learner.entropy_loss_weight
     to_device = x -> send_to_device(device(AC), x)
 
-    states = select_last_dim(t[:state], 1:n)
-    states_flattened = flatten_batch(states) |> to_device # (state_size..., n_thread * update_freq)
+    S = t[:state] |> to_device
+    states = select_last_dim(S, 1:n)
```

Comment:
Not related to this PR specifically, but why do we do `select_last_dim` with `1:n`, shouldn't that be the whole range?

Reply:
Because in A2C, only the current state is considered ;)
The actor part calculates the probability of the action taken at the current states, and the critic part calculates the estimated reward of the current states. (If I still remember A2C correctly...)

Reply:
But shouldn't this pretty much be a noop? In my understanding we have `select_last_dim(S, 1:n) == S`

Reply:
>  it is the length of t but does that mean it is the length of the last dimension of S?

n is the number transitions in a trajectory. For a trajectory of type CircularSARTTrajectory, the (last dimension )length of state and action field is one step longer than the (last dimension ) length of reward and terminal fields. The extra one means next state and action.

Reply:
https://github.com/JuliaReinforcementLearning/ReinforcementLearning.jl/blob/732ecfc2615010189d9c2d402c313351b2591caf/src/ReinforcementLearningCore/src/policies/agents/trajectories/trajectory.jl#L542

Reply:
Okay, now I understand. All is good :)